src/packageRequester.ts

```diff
@@ -486,3 +493,23 @@ async function fetcher (
   }
   return await fetch(resolution, target, opts)
 }
+
+async function getSideEffectsCache (storePath: string, id: string, useCache: boolean): Promise<Map<string, string>> {
+  const map = new Map()
+  if (!useCache) {
+    return map
+  }
+
+  const cacheRoot = path.join(storePath, 'side-effects-cache')
+  const dirContents = await fs.readdir(cacheRoot)
+  const isDirectory = async (source: string) => (await fs.lstat(source)).isDirectory()
+  const subDirs = await Promise.all(dirContents.map((content) => path.join(cacheRoot, content)).filter(isDirectory))
+  await Promise.all(subDirs.map(async (dir) => {
+    const nodeMajor = path.basename(dir)
+    const maybeCacheDir = path.join(dir, id)
+    if (await fs.exists(maybeCacheDir)) {
+      map[nodeMajor] = maybeCacheDir
+    }
+  }))
+  return map
```

Comment:
It will be better to put the side effects cache to the same folder where the rest of the resources are, so into `<store>/<pkgid>/side_effects/node<major version>/`

This will make it easier to prune the store.